app/server/lib/DocManager.ts

```diff
@@ -733,3 +739,41 @@ export class DocManager extends EventEmitter {
 function extname(fpath: string): string {
   return path.extname("X" + fpath);
 }
+
+async function updateDocumentAttachmentStoreSettingToValidValue(fname: string, provider: IAttachmentStoreProvider) {
+  return updateDocumentSettingsInPlace(fname, (oldSettings) => {
+    const attachmentStoreId = oldSettings?.attachmentStoreId;
+    if (!attachmentStoreId || provider.storeExists(attachmentStoreId)) {
+      return oldSettings;
+    }
+    const newStoreLabel = getConfiguredStandardAttachmentStore();
+    const newStoreId = newStoreLabel && provider.getStoreIdFromLabel(newStoreLabel);
+
+    return {
+      ...oldSettings,
+      attachmentStoreId: newStoreId,
+    };
+  });
+}
+
+// Updates the document's settings (_grist_DocInfo.docSettings) without loading the document.
+async function updateDocumentSettingsInPlace(
+  fname: string,
+  makeChanges: (oldSettings: DocumentSettings | undefined) => DocumentSettings | undefined
+) {
+  const db = await SQLiteDB.openDBRaw(fname, OpenMode.OPEN_EXISTING);
+  const columns = await db.all("PRAGMA table_info(_grist_DocInfo)");
+  if (!columns.some(column => column.name === 'documentSettings')) {
+    return;
+  }
+  const results = await db.all('SELECT id, schemaVersion, documentSettings FROM _grist_DocInfo');
```

Comment:
`_grist_DocInfo` is a singleton, so `db.get` might read more simply?

Reply:
This kind of thing always makes me nervous (as it breaks if the singleton assumption is ever violated), but it seems like a reasonable simplification here (since presumably the rest of the code would also break).